secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/dcr/request/DCRRegistrationRequestJwtSignatureValidatorJwks.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.dcr.request;
+
+import java.security.SignatureException;
+
+import org.apache.ivy.util.StringUtils;
+import org.forgerock.http.protocol.Response;
+import org.forgerock.http.protocol.Status;
+import org.forgerock.json.JsonException;
+import org.forgerock.json.JsonValue;
+import org.forgerock.json.jose.jwk.JWKSet;
+import org.forgerock.json.jose.jws.SignedJwt;
+import org.forgerock.json.jose.jwt.JwtClaimsSet;
+import org.forgerock.util.promise.Promise;
+import org.forgerock.util.promise.Promises;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.forgerock.sapi.gateway.dcr.request.DCRSignatureValidationException.ErrorCode;
+
+import com.forgerock.sapi.gateway.jws.JwtSignatureValidator;
+import com.forgerock.sapi.gateway.trusteddirectories.TrustedDirectory;
+
+/**
+ * Class used to validate a registration request jwt signature against a JWKS embedded in the Software Statement
+ */
+public class DCRRegistrationRequestJwtSignatureValidatorJwks implements DCRRegistrationRequestValidator {
+
+    private static final Logger log = LoggerFactory.getLogger(DCRRegistrationRequestJwtSignatureValidatorJwks.class);
+    private final JwtSignatureValidator jwtSignatureValidator;
+
+    /**
+     * Constructor
+     * @param jwtSignatureValidator service that is used to validate a SignedJwt against a JWKSet
+     */
+    public DCRRegistrationRequestJwtSignatureValidatorJwks(JwtSignatureValidator jwtSignatureValidator) {
+        this.jwtSignatureValidator = jwtSignatureValidator;
```

Comment:
I'm sure this is just oversight as you are doing it in other places, but it is always a good practice to valid required dependencies are supplied in the constructor so we can fail fast here.